sticht/rollbacks/base.py

```diff
@@ -0,0 +1,235 @@
+import abc
+from typing import Any
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from sticht.rollbacks.slo import SLOWatcher
+from sticht.rollbacks.slo import watch_slos_for_service
+from sticht.slack import Emoji
+from sticht.slack import SlackDeploymentProcess
+
+
+class RollbackSlackDeploymentProcess(SlackDeploymentProcess, abc.ABC):
+    def __init__(self) -> None:
+        super().__init__()
+        self.slo_watchers: Optional[List[SLOWatcher]] = None
+        self.metric_watchers: Optional[List[Any]] = None
```

Comment:
we have an constructor now so that we can avoid doing some getattr magic below to see if we've initialized our watcher list(s)